### DISCUSSION TITLE
Facing issue `header file not found` while running testcontainers on latest MacOS Darwin versions. Any help is appreciated!

### DISCUSSION THREAD

**Discussion by iamQ0** [testcontainers/testcontainers-go#2740] [General]
Hey folks, so I'm facing an issue with few of the testcontainers dependencies, `fsevents` and `go-m1cpu` wherein they are failing to build with error `header file not found`. Errors attached below: 

```
# github.com/fsnotify/fsevents
../../go/pkg/mod/github.com/fsnotify/fsevents@v0.2.0/wrap.go:7:10: fatal error: 'CoreServices/CoreServices.h' file not found
#include <CoreServices/CoreServices.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
# github.com/shoenig/go-m1cpu
../../go/pkg/mod/github.com/shoenig/go-m1cpu@v0.1.6/cpu.go:8:11: fatal error: 'IOKit/IOKitLib.h' file not found
 #include <IOKit/IOKitLib.h>
 ```
 `go-m1cpu` is direct testcontainers dependency but `fsevents` is used by docker compose API which is being imported in testcontainers-go. 
 
Below is my machine info:
```
❯ uname -v
Darwin Kernel Version 23.6.0: Mon Jul 29 21:14:30 PDT 2024; root:xnu-10063.141.2~1/RELEASE_ARM64_T6030
```

I have tried updating/installing XCode and CLTs but still it fails to build.
```
macOS: 14.6.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.4
Rosetta 2: false
```
Possible issue being : 
In the newer versions of MacOS the headers files have been moved from`/usr/include` to Xcode bundle under `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk`. 
So fsevents is failing to build since its possibly trying to find the files in the original location. 

This changed has been applied for new Mac and Xcode versions 10 and above.
Check [release notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-10-release-notes#Command-Line-Tools).

> If you are the maintainer of such software, we encourage you to update your project to work with the SDK or file a bug report for issues that are preventing you from doing so

I have even tried setting appropriate `CGO_CFLAGS` and `CGO_LDFLAGS` before running : 
```
CGO_CFLAGS="-isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk -I/usr/include" 

CGO_LDFLAGS="-isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk -L/usr/lib" 
```
but that doesn't seem to resolve the issue as well. 
 
If any of you have come across such related errors, please lemme know!

**Comment by mdelapenya** [comment]
I think this issue was pushed upstream to https://github.com/fsnotify/fsevents/issues/65. I guess we can close it and reopen if needed.

Thanks!
